cmd/cca/verify_as_relyingparty.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2023 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+package cca
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/veraison/apiclient/verification"
+	"github.com/veraison/ccatoken"
+	"github.com/veraison/evcli/common"
+)
+
+var (
+	relyingPartyTokenFile *string
+	relyingPartyAPIURL    *string
+)
+
+var (
+	relyingPartyVeraisonClient common.IVeraisonClient = &verification.ChallengeResponseConfig{}
+	relyingPartyCmd                                   = NewRelyingPartyCmd(common.Fs, relyingPartyVeraisonClient)
+)
+
+func NewRelyingPartyCmd(fs afero.Fs, veraisonClient common.IVeraisonClient) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "relying-party",
+		Short: "Emulate a Relying Party",
+		Long: `This command implements the "relying party mode" of a
+challenge-response interaction, where the relying party was the original
+challenger, and therefore the nonce is provided by the caller implicitly in an
+already well-formed CCA attestation token, possibly produced by a
+previous invocation to "evcli cca create" command.
+
+	evcli cca verify-as relying-party \
+	              --api-server=https://veraison.example/challenge-response/v1/newSession \
+	              --token=cca-token.cbor
+
+	`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			token, err := afero.ReadFile(fs, *relyingPartyTokenFile)
+			if err != nil {
+				return err
+			}
+
+			var e ccatoken.Evidence
+
+			if err = e.FromCBOR(token); err != nil {
+				return err
+			}
+
+			nonce, err := e.RealmClaims.GetChallenge()
+			if err != nil {
+				return err
+			}
+
+			if err = veraisonClient.SetNonce(nonce); err != nil {
+				return err
+			}
+
+			if err = veraisonClient.SetSessionURI(*relyingPartyAPIURL); err != nil {
+				return err
+			}
+
+			eb := relyingPartyEvidenceBuilder{Token: token, Nonce: nonce}
+			if err = veraisonClient.SetEvidenceBuilder(eb); err != nil {
+				return err
+			}
+
+			veraisonClient.SetDeleteSession(true)
+
+			attestationResults, err := veraisonClient.Run()
+			if err != nil {
+				return err
+			}
```

Comment:
same comment about context-free `return err` here as above

Reply:
Added a bit of description to understand which method is failing now!

Reply:
I might be looking in the wrong place but I can't see any change in L52,57,61,... here?

Reply:
I think the set from the API itself is verbose so we are ok.

For Run specifically I have changed it, as Run has particularly not very good return error description

Reply:
I think I didn't made my point clear.  It's not that the error strings returned from the library calls are bad/poor and therefore we need to make them better. 
apiclient has pretty good reporting, but even if it hadn't that should be fixed in apiclient rather than patched here...  The thing I'm trying to point out is that the calling sites should add context to the error reported by the library to make clearer where the error occurred in the main flow.

Reply:
Rather than going in circles, I can do a proposal.

Reply:
see [4db22e3](https://github.com/veraison/evcli/pull/22/commits/4db22e39ce10056683b2f3fed1eedc89cfba055a)